src/StreamJsonRpc/SkipClrVisibilityChecks.cs

```diff
@@ -131,6 +134,25 @@ internal void SkipVisibilityChecksFor(AssemblyName assemblyName)
             }
         }
 
+        private static IEnumerable<TypeInfo> ThisAndBaseTypes(TypeInfo startingPoint)
+        {
+            if (startingPoint.IsInterface)
+            {
+                yield return startingPoint.GetTypeInfo();
+                foreach (Type iface in startingPoint.GetInterfaces())
+                {
+                    yield return iface.GetTypeInfo();
+                }
+            }
+            else
+            {
+                for (TypeInfo? t = startingPoint.GetTypeInfo(); t is not null && t != typeof(object).GetTypeInfo(); t = t.BaseType?.GetTypeInfo())
+                {
+                    yield return t;
```

Comment:
Do we have to recurse into interfaces implemented by these base types or do we only care about the interfaces declared directly by the original starting point?

Reply:
You know, I copied this code from another place in the library without thinking. This code will never be hit because proxies are always based on *interfaces*, not classes. I'll delete.